contracts/dev-utils/contracts/src/LibDydxBalance.sol

```diff
@@ -0,0 +1,418 @@
+            if (accountIdx != withdraw.accountIdx) {
+                continue;
+            }
+            // Compute the deposit/collateralization rate, which is the rate at
+            // which (USD) value is added to the account across all markets.
+            int256 dd = 0;
+            for (uint256 i = 0; i < info.actions.length - 1; ++i) {
+                IDydxBridge.BridgeAction memory deposit = info.actions[i];
+                assert(deposit.actionType == IDydxBridge.BridgeActionType.Deposit);
+                if (deposit.accountIdx == accountIdx) {
+                    dd = D18.add(
+                        dd,
+                        _getActionRateValue(
+                            info,
+                            deposit
+                        )
+                    );
+                }
+            }
+            // Compute the borrow/withdraw rate, which is the rate at which
+            // (USD) value is deducted from the account.
+            int256 db = _getActionRateValue(
+                info,
+                withdraw
+            );
+            // If the deposit to withdraw ratio is >= the minimum collateralization
+            // ratio, then we will never become insolvent at these prices.
+            if (D18.div(dd, db) >= minCr) {
+                continue;
+            }
+            // The collateralization ratio for this account, parameterized by
+            // `t` (maker amount), is given by:
+            //      `cr = (supplyValue + t * dd) / (borrowValue + t * db)`
+            // Solving for `t` gives us:
+            //      `t = (supplyValue - cr * borrowValue) / (cr * db - dd)`
+            int256 t = D18.div(
+                D18.sub(supplyValue, D18.mul(minCr, borrowValue)),
+                D18.sub(D18.mul(minCr, db), dd)
+            );
+            solventMakerAmount = LibSafeMath.min256(
+                solventMakerAmount,
+                // `t` is in maker token units, so convert it to maker wei.
+                _toWei(info.makerTokenAddress, uint256(D18.clip(t)))
+            );
+        }
+    }
+
+    /// @dev Create a `BalanceCheckInfo` struct.
+    /// @param order An order with a `DydxBridge` maker asset.
+    /// @param dydx The address of the Dydx contract.
+    /// @return info The `BalanceCheckInfo` struct.
+    function _getBalanceCheckInfo(LibOrder.Order memory order, address dydx)
+        private
+        pure
+        returns (BalanceCheckInfo memory info)
+    {
+        bytes memory rawBridgeData;
+        (, info.makerTokenAddress, info.bridgeAddress, rawBridgeData) =
+            LibAssetData.decodeERC20BridgeAssetData(order.makerAssetData);
+        info.dydx = IDydx(dydx);
+        info.makerAddress = order.makerAddress;
+        if (order.takerAssetData.readBytes4(0) == IAssetData(0).ERC20Token.selector) {
+            (, info.takerTokenAddress) =
+                LibAssetData.decodeERC20AssetData(order.takerAssetData);
+        }
```

Comment:
I wonder if we should revert if we cannot extract the taker token address. O/w it seems the other functions may not perform as expected. Failing here may be more graceful.

Reply:
Actually, this already would revert but I think we should do the opposite and not revert. The other functions do check if the taker token is null.